shell/platform/android/external_view_embedder/surface_pool.cc

```diff
@@ -31,7 +31,13 @@ std::shared_ptr<OverlayLayer> SurfacePool::GetLayer(
 
   intptr_t gr_context_key = reinterpret_cast<intptr_t>(gr_context);
   // Allocate a new surface if there isn't one available.
-  if (available_layer_index_ >= layers_.size()) {
+  bool needsLayer;
+  {
+    std::lock_guard lock(layers_mutex_);
+    needsLayer = available_layer_index_ >= layers_.size();
```

Comment:
Is there any risk that the `layers_` list may be modified between this check and the time when the layer is used?

The lock may need to be held during the entire operation.  Or it may need an approach such as:
* with the lock held:
  * obtain a layer if available and remove it from `layers_`
* if no layer was available, then create a layer
* with the lock held:
  * set up the layer and insert it into `layers_`

Reply:
I changed it in https://github.com/flutter/engine/pull/31464/commits/117787a91479644259b0a38decc7c28e79308042, so the lock is held during the entire operation. PTAL